src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/util/RcaResponseUtil.java

```diff
@@ -0,0 +1,119 @@
+
+public class RcaResponseUtil {
+
+  private static final String NULL = "null";
+
+  public static String convertRcaRecordListIntoJson(String rcaName, List<Record> recordList, Set<String> tableNames) {
+    com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.response.RcaResponse rcaResponse = null;
+    Map<String, NodeSummaryResponse> nodeSummaryResponseMap = new HashMap<>();
+    for (Record record : recordList) {
+      if (rcaResponse == null) {
+        rcaResponse = getRcaResponse(rcaName, record, tableNames);
+      }
+      if (tableNames.contains(HOT_NODE_SUMMARY_TABLE)) {
+        String nodeId = getStringFieldValue(record, HotNodeSummary.SQL_SCHEMA_CONSTANTS.NODE_ID_COL_NAME);
+        NodeSummaryResponse nodeSummaryResponse = nodeSummaryResponseMap.getOrDefault(nodeId,
+                getNodeSummary(nodeId, record));
+        if (!nodeSummaryResponseMap.containsKey(nodeId)) {
+          nodeSummaryResponseMap.put(nodeId, nodeSummaryResponse);
+          rcaResponse.addNodeSummaryResponse(nodeSummaryResponse);
+        }
+        if (tableNames.contains(HOT_RESOURCE_SUMMARY_TABLE)) {
+          nodeSummaryResponse.addResource(getResourceSummaryResponse(record));
+        }
+      }
+    }
+    return rcaResponse == null ? "{}" : rcaResponse.toString();
+  }
+
+  private static ResourceSummaryResponse getResourceSummaryResponse(Record record) {
+    String resourceName = getStringFieldValue(record,
+            HotResourceSummary.SQL_SCHEMA_CONSTANTS.RESOURCE_TYPE_COL_NAME);
+    String unitType = getStringFieldValue(record,
+            HotResourceSummary.SQL_SCHEMA_CONSTANTS.UNIT_TYPE_COL_NAME);
+    Double threshold = getDoubleFieldValue(record,
+            HotResourceSummary.SQL_SCHEMA_CONSTANTS.THRESHOLD_COL_NAME);
+    Double average = getDoubleFieldValue(record,
+            HotResourceSummary.SQL_SCHEMA_CONSTANTS.AVG_VALUE_COL_NAME);
+    Double actual = getDoubleFieldValue(record,
+            HotResourceSummary.SQL_SCHEMA_CONSTANTS.VALUE_COL_NAME);
+    Double maximum = getDoubleFieldValue(record,
+            HotResourceSummary.SQL_SCHEMA_CONSTANTS.MAX_VALUE_COL_NAME);
+    Double minimum = getDoubleFieldValue(record,
+            HotResourceSummary.SQL_SCHEMA_CONSTANTS.MIN_VALUE_COL_NAME);
+
+    return new ResourceSummaryResponse(resourceName, unitType, threshold, actual, average, minimum, maximum);
+  }
+
+  private static String getStringFieldValue(Record record, String fieldName) {
+    Object value = record.getValue(fieldName);
+    return value == null ? NULL : value.toString();
+  }
+
+  private static Double getDoubleFieldValue(Record record, String fieldName) {
+    String value = getStringFieldValue(record, fieldName);
+    return value.equals(NULL) ? null : Double.parseDouble(value);
+  }
+
+  private static Integer getIntegerFieldValue(Record record, String fieldName) {
+    String value = getStringFieldValue(record, fieldName);
+    return value.equals(NULL) ? null : Integer.parseInt(value);
+  }
```

Comment:
We don't have to create a method for each data type. Can we use this API in JOOQ record instead ? 

get​(String fieldName, Class<? extends T> type)
--

Reply:
Removed methods for each data type.